Payment/Program.cs

```diff
@@ -1,2 +1,97 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿namespace Payment;
+
+public interface IPaymentMethod
+{
+    void CalculateDiscounts(decimal amount);
+}
+
+public class PayPal : IPaymentMethod
+{
+    public void CalculateDiscounts(decimal amount)
+    {
+        Console.WriteLine($"total amount: ${amount}");
+    }
+}
+public class CreditCard : IPaymentMethod
+{
+    public void CalculateDiscounts(decimal amount)
+    {
+        Console.WriteLine($"Applying 2% fee for credit card payment.'\n' Processing credit card payment of ${amount * 1.02m}");
+
+    }
+}
+//Adapter
+public class CryptoCurrencyAdapter : IPaymentMethod
+{
+    private readonly CryptoCurrencyThirdParty _cryptoCurrencyThirdParty;
+    public CryptoCurrencyAdapter(CryptoCurrencyThirdParty cryptoCurrencyThirdParty)
+    {
+        _cryptoCurrencyThirdParty = cryptoCurrencyThirdParty;
+    }
+    public void CalculateDiscounts(decimal amount)
+    {
+        _cryptoCurrencyThirdParty.ProcessCryptoPayment(amount);
+    }
+}
+//NullObject
+public class UnknownPayment : IPaymentMethod
+{
+
+    public void CalculateDiscounts(decimal amount)
+    {
+        Console.WriteLine($"No adjustment for the selected payment method.'\n'Payment method not supported.Unable to process ${amount}.");
+
+    }
+}
+
+//Third Party Class
+public class CryptoCurrencyThirdParty
+{
+    public void ProcessCryptoPayment(decimal amount)
+    {
+        Console.WriteLine($"Processing cryptocurrency payment of ${amount} via third-party library");
+
+    }
+}
+//Factory
+public class PaymentFactory
+{
+    public static IPaymentMethod GetPaymentMethod(string paymentMethod)
+    {
```

Comment:
great simple factory method. 

now you can use it with dictionary. if you want to practice more.